langwatch/src/optimization_studio/components/Evaluate.tsx

```diff
@@ -413,48 +430,80 @@ export function EvaluateModalContent({
 const DatasetSplitSelect = ({
   field,
   options,
+  total,
 }: {
   field: ControllerRenderProps<EvaluateForm, "evaluateOn">;
   options: DatasetSplitOption[];
+  total?: number;
 }) => {
   const datasetSplitCollection = createListCollection({
     items: options,
   });
 
   return (
-    <Select.Root
-      {...field}
-      collection={datasetSplitCollection}
-      value={field.value?.value ? [field.value.value] : []}
-      onChange={undefined}
-      onValueChange={(change) => {
-        const selectedOption = options.find(
-          (option) => option.value === change.value[0]
-        );
-        field.onChange({
-          target: {
-            name: field.name,
-            value: selectedOption,
-          },
-        });
-      }}
-      width="100%"
-    >
-      <Select.Trigger>
-        <Select.ValueText placeholder="Select dataset split" />
-      </Select.Trigger>
-      <Select.Content zIndex="popover">
-        {options.map((option) => (
-          <Select.Item item={option} key={option.value}>
-            <VStack align="start" width="full">
-              <Text>{option.label}</Text>
-              <Text fontSize="13px" color="gray.500">
-                {option.description}
-              </Text>
-            </VStack>
-          </Select.Item>
-        ))}
-      </Select.Content>
-    </Select.Root>
+    <VStack width="100%" gap={2}>
+      <Select.Root
+        {...field}
+        collection={datasetSplitCollection}
+        value={field.value?.value ? [field.value.value] : []}
+        onChange={undefined}
+        onValueChange={(change) => {
+          const selectedOption = options.find(
+            (option) => option.value === change.value[0]
+          );
+          field.onChange({
+            target: {
+              name: field.name,
+              value: selectedOption,
+            },
+          });
+        }}
+        width="100%"
+      >
+        <Select.Trigger>
+          <Select.ValueText placeholder="Select dataset split" />
+        </Select.Trigger>
+        <Select.Content zIndex="popover">
+          {options.map((option) => (
+            <Select.Item item={option} key={option.value}>
+              <VStack align="start" width="full">
+                <Text>{option.label}</Text>
+                <Text fontSize="13px" color="gray.500">
+                  {option.description}
+                </Text>
+              </VStack>
+            </Select.Item>
+          ))}
+        </Select.Content>
+      </Select.Root>
+      {field.value?.value === "specific" && (
+        <Input
+          type="text"
+          placeholder={`Enter row index (0-${(total ?? 0) - 1})`}
+          value={
+            field.value?.datasetEntry !== undefined
+              ? String(field.value.datasetEntry)
+              : ""
```

Comment:
I like to avoid ternary if possible
```suggestion
           String(field.value.datasetEntry ?? "")
```